lib/src/git_backend.rs

```diff
@@ -1949,47 +2010,55 @@ mod tests {
         let read_commit = backend.read_commit(&read_commit_id).block_on().unwrap();
         assert_eq!(read_commit, commit);
         let git_commit = git_repo
-            .find_commit(Oid::from_bytes(read_commit_id.as_bytes()).unwrap())
+            .find_commit(gix::ObjectId::from_bytes_or_panic(
+                read_commit_id.as_bytes(),
+            ))
             .unwrap();
-        let git_tree = git_repo.find_tree(git_commit.tree_id()).unwrap();
+        let git_tree = git_repo.find_tree(git_commit.tree_id().unwrap()).unwrap();
         assert!(git_tree
             .iter()
-            .filter(|entry| entry.name() != Some("README"))
-            .all(|entry| entry.filemode() == 0o040000));
+            .filter(|entry| match entry {
```

Comment:
nit: it's probably better to insert `.map(Result::unwrap)` here instead of filtering out errors.